azure/src/test/java/org/apache/iceberg/azure/AzurePropertiesTest.java

```diff
@@ -71,6 +78,49 @@ public void testWithSasToken() {
     verify(clientBuilder, never()).credential(any(StorageSharedKeyCredential.class));
   }
 
+  @Test
+  public void testWithRefreshCredentialsEndpoint() {
+    try (var providerMockedConstruction = mockConstruction(VendedAdlsCredentialProvider.class);
```

Comment:
I'd suggest to give all of these some normal names, such as `credendialProvider` instead of `providerMockedConstruction`